File: Selenium/Activities/Activity4.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.firefox.webdriver import WebDriver
from selenium.webdriver.support import expected_conditions
from webdriver_manager.firefox import GeckoDriverManager

# setup Firefoxdriver with WebDriverManager
service = FirefoxService(GeckoDriverManager().install())

# start the driver
with webdriver.Firefox(service=service) as driver:
    # Access the URL
    driver.get("https://v1.training-support.net/selenium/target-practice")
    # print header of the page
    print("Title of the home page is : ", driver.title)
    # find 3rd header of the page and print the text
    thirdheader = driver.find_element(By.XPATH,"//h3[@id='third-header']")
    print("Text of the third header is : ", thirdheader.text)
    # find 5th header of the page and print its color
    fifthHeader = driver.find_element(By.XPATH,"//h5[contains(text(), 'Fifth header')]")
    print("Color of the fifth header is: " + fifthHeader.value_of_css_property("color"))
    # locate the violet button and print its classed
    violetButton = driver.find_element(By.XPATH,"//button[contains(text(), 'Violet')]")
    print("Class of violet button is: " + violetButton.get_attribute("class"))
    # locate the grey button and print text
    greyButton = driver.find_element(By.XPATH,"//button[@class = 'ui grey button']")
    print("Text of grey button is: " + greyButton.text)
    # close the browser
    driver.close()

File: Selenium/Activities/Activity4.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity4 {
    public static void main(String[] args) throws InterruptedException {
        //setting up system property
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE,"/dev/null");
        //setting up Webdrivermanager
        WebDriverManager.firefoxdriver().setup();
        //Initializing webdriver
        WebDriver driver = new FirefoxDriver();
        //Access the URL
        driver.get("https://v1.training-support.net/selenium/target-practice");
        //print header of the page
        System.out.println("Title of the home page is : "+driver.getTitle());
        //find 3rd header of the page and print the text
        WebElement thirdHeader = driver.findElement(By.xpath("//h3[@id='third-header']"));
        System.out.println("Text of the third header is : "+thirdHeader.getText());
        //find 5th header of the page and print its color
        WebElement fifthHeader = driver.findElement(By.xpath("//h5[contains(text(), 'Fifth header')]"));
        System.out.println("Color of the fifth header is: "+fifthHeader.getCssValue("color"));
        //locate the violet button and print its classed
        WebElement violetButton = driver.findElement(By.xpath("//button[contains(text(), 'Violet')]"));
        System.out.println("Class of violet button is: "+violetButton.getAttribute("class"));
        //locate the grey button and print text
        WebElement greyButton = driver.findElement(By.xpath("//button[@class = 'ui grey button']"));
        System.out.println("Text of grey button is: "+greyButton.getText());
        //close the browser
        driver.close();

} }
